### PR TITLE
Refactor Resize/Upsample implementation to reduce binary size. 

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Refactor a few parts of the Resize/Upsample implementation to split out type agnostic code from templatized code.

**Motivation and Context**
Reduce binary size by ~17KB (Windows release build).

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/upsample.cc [hunk 1]
```diff
@@ -926,30 +1106,28 @@ Status Upsample<T>::BaseCompute(OpKernelContext* context,
                           output_depth, output_height, output_width,
                           is_3D ? scales[0] : scales[2], is_3D ? scales[1] : scales[3],
                           is_3D ? scales[2] : scales[4], roi, use_extrapolation_, extrapolation_value_,
-                          X->template Data<T>(), Y->template MutableData<T>(), alloc, get_original_coordinate_, 
+                          X->Data<T>(), Y->MutableData<T>(), alloc, get_original_coordinate_,
                           output_height * output_width > 64 ? context->GetOperatorThreadPool() : nullptr);
         return Status::OK();
       } else {
         // User shouldn't hit this as the check has been performed in ScalesValidation()
         std::ostringstream oss;
```
Comment (edgchen1) [inline onnxruntime/core/providers/cpu/tensor/upsample.cc:1114]: >oss [](http://example.com/codeflow?start=27&length=3)

`oss` not needed anymore

**Comment by edgchen1** [comment]
> #include <sstream>

is <sstream> still needed?

---
Refers to: onnxruntime/core/providers/cpu/tensor/upsample.cc:7 in 57bede6. [](commit_id = 57bede6b437f4a827e00e0a307b69484afe15c35, deletion_comment = False)

File: onnxruntime/core/providers/cpu/tensor/upsample.cc [hunk 2]
```diff
@@ -227,27 +254,71 @@ Status UpsampleNearest(const T* input,
...
+}
+
+/*
+
+// This is a generic upsample in linear mode for N-D tensor.
+// But what's the correct behavior for linear mode is not clear right now.
+// this function is not enabled yet.
+// this function is not tested for opset 11 changes yet
+
+static Status UpsampleLinearImpl(const std::function<void(size_t, size_t, float)>& apply,
```
Comment (edgchen1) [inline onnxruntime/core/providers/cpu/tensor/upsample.cc:315]: maybe keep it uncommented out? it's easier to overlook commented out code when refactoring

**Comment by edgchen1** [comment]
> #include "core/providers/cpu/tensor/upsample.h"

can this be included first?

---
Refers to: onnxruntime/core/providers/cpu/tensor/upsample.cc:6 in 57bede6. [](commit_id = 57bede6b437f4a827e00e0a307b69484afe15c35, deletion_comment = False)

**Review by edgchen1 — APPROVED** [review]

File: onnxruntime/core/providers/cpu/tensor/upsample.cc (hunk 2 above)
Comment (skottmckay) [reply onnxruntime/core/providers/cpu/tensor/upsample.cc:315]: with the refactoring not commenting it out leads to a build warning/failure. 

given it has been years without this being enabled I'm not sure we need it. at least it's easier to find if it's here and commented out.

**Review by gwang-msft — APPROVED** [review]

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**